src/oop/special_class_method/hard/q5/MainTest.java

```diff
@@ -0,0 +1,315 @@
+    public void roadClassHasCorrectConstructor(int value1, String value2, int value3) throws Throwable {
+        Object[][] arguments = {
+                {value1, int.class},
+                {value2, String.class},
+                {value3, int.class}
+        };
+        Class<?>[] constructorClasses = {int.class, String.class, int.class};
+        Object classInstance = testClass.createInstance(arguments);
+        String incorrectIntValueMessage =
+                "Your " + className + " constructor does not initialize the " + intFieldName + " attribute to the correct value.";
+        String incorrectStringValueMessage =
+                "Your " + className + " constructor does not initialize the " + stringFieldName + " attribute to the correct value.";
+        String incorrectInt2ValueMessage =
+                "Your " + className + " constructor does not initialize the " + int2FieldName + " attribute to the correct value.";
+        String incorrectConstructorVisibilityModifier =
+                "Your " + className + " constructor does not have the correct visibility modifier.";
+        _assertEquals(value1, testClass.getFieldValue(classInstance, intFieldName),
+                incorrectIntValueMessage);
+        _assertEquals(value2, testClass.getFieldValue(classInstance, stringFieldName),
+                incorrectStringValueMessage);
+        _assertEquals(value3, testClass.getFieldValue(classInstance, int2FieldName),
+                incorrectInt2ValueMessage);
+        assertTrue(testClass.hasModifier(constructorClasses, "public"), incorrectConstructorVisibilityModifier);
+    }
+
+    private static Stream<Arguments> twoIntInputProvider() {
+        return Stream.of(
+                Arguments.of(4, 3),
+                Arguments.of(3, 6),
+                Arguments.of(1, 4),
+                Arguments.of(6, 1)
+
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoIntInputProvider")
+    public void roadClassHasCorrectSetLanesMethod(int initialValue, int setValue) throws Throwable {
+        Object[][] arguments = {
+                {initialValue, int.class},
+                {"Initial String", String.class},
+                {2, int.class}
+        };
+        Object classInstance = testClass.createInstance(arguments);
+        Object[][] setSizeArguments = {
+                {setValue, int.class}
+        };
+        String[] setMethodModifiers = {"public"};
+        String incorrectSetMethodMessage = String.join(" ",
+                "Your", setIntMethodName, "method does not correctly set the value of the", intFieldName, "attribute.");
+        Object setMethodOutput = testClass.callMethod(setIntMethodName, setSizeArguments, setMethodModifiers, classInstance);
+        _assertEquals(setValue, testClass.getFieldValue(classInstance, intFieldName), incorrectSetMethodMessage);
+        assertNull(setMethodOutput, String.join(" ", "Your", setIntMethodName, "method should not return any output"));
```

Comment:
I think it is better to use the return type assert instead of the assertNull here